### PR TITLE
BUG: loc.setitem with expansion expanding rows

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

discovered on branch that has dataframe always go through split_path

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -1684,6 +1684,13 @@ def _setitem_with_indexer_split_path(self, indexer, value):
                 for loc, v in zip(ilocs, value):
                     self._setitem_single_column(loc, v, pi)
 
+            elif len(ilocs) == 1 and com.is_null_slice(pi) and len(self.obj) == 0:
```
Comment (jreback) [inline pandas/core/indexing.py:1687]: hmm this case is handled above no (before the split path)? (unless you changed the code)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -953,6 +953,13 @@ def test_loc_setitem_empty_append(self):
         df.loc[:, "x"] = data
         tm.assert_frame_equal(df, expected)
 
+        # same thing, but with mixed dtypes
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:956]: make a separate test

File: pandas/core/indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py:1687]: in the homogeneous-dtype case this goes through the non-split path.  With mixed dtypes we go through split path, and in master fall through to raise a few lines down

File: pandas/tests/indexing/test_loc.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_loc.py]: updated+green

**Comment by jbrockmendel** [comment]
@phofl do you have any other upcoming PRs that affect `_setitem_with_indexer_split_path`?  This and #37931 are my last two before im ready to start in on the always-split-path change, and id like to get any bugfixes+tests in place before that.

**Comment by phofl** [comment]
I think only #37749, but will have a closer look. At least nothing which is not already opened

**Comment by jreback** [comment]
is this a user facing change? (if so pls add a whatsnew), also merge master and ping on green.

**Comment by jreback** [comment]
is this a user facing change? (if so pls add a whatsnew), also merge master and ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**